arches_modular_reports/app/utils/nodegroup_tile_data_utils.py

```diff
@@ -180,17 +180,31 @@ def get_sorted_filtered_tiles(
     *,
```

Comment:
this method is starting to feel overloaded, but I just don't see a graceful way to simplify this or split this out into multiple methods.